File: tests/perf/test-hex-decode.js
if (typeof print !== 'function') { print = console.log; }

function test() {
    var tmp1 = [];
    var tmp2 = [];
    var i, n;

    print('build');
    for (i = 0; i < 1024; i++) {
        tmp1.push(Math.floor(Math.random() * 16).toString(16))
    }
    tmp1 = tmp1.join('');
    for (i = 0; i < 1024; i++) {
        tmp2.push(tmp1);
    }
    tmp2 = tmp2.join('');

    print(tmp2.length);
    print('run');
    for (i = 0; i < 10000; i++) {
        // Assigning to 'res' avoids garbage collection of result; this is
        // intentional to avoid mixing string intern performance to the test.
        var res = Duktape.dec('hex', tmp2);
    }
}

try {
    test();
} catch (e) {
    print(e.stack || e);
    throw e;
}

File: tests/perf/test-hex-decode.py
import math
import random

def test():
	tmp1 = []
	tmp2 = []

	print('build')
	for i in xrange(1024):
		tmp1.append('%x' % math.floor(random.random() * 16))
	tmp1 = ''.join(tmp1)
	for i in xrange(1024):
		tmp2.append(tmp1)
	tmp2 = ''.join(tmp2)

	print(len(tmp2))
	print('run')
	for i in xrange(10000):
		res = tmp2.decode('hex')

test()
